### PR TITLE
Modal simulation driver update

### PR DESCRIPTION
**Pull request by kmokstad**
Adding structural damping parameters, RHS and solution vectors as members in class SIMmodal.
Some of the virtual methods then no longer need to be virtual.

### CODE REVIEW

File: src/SIM/SIMmodal.C [hunk 1]
```diff
@@ -78,46 +100,57 @@ bool SIMmodal::swapSystem (AlgEqSystem*& sys, SAM*& sam)
...
     }
 
-  return true;
+  return sol;
+}
+
+
+const Vector& SIMmodal::expandedSolution (int idx) const
+{
+  if (idx >= 0 && idx < (int)sol.size())
```
Comment (akva2) [inline src/SIM/SIMmodal.C:149]: c++ cast

File: src/SIM/SIMmodal.C [hunk 2]
```diff
@@ -67,6 +73,22 @@ SIMmodal::~SIMmodal ()
 }
 
 
+bool SIMmodal::parseParams (const tinyxml2::XMLElement* elem)
+{
+  if (parsed)
+    IFEM::cout <<"\t(skipped)"<< std::endl;
+  else if (!strcasecmp(elem->Value(),"newmarksolver"))
```
Comment (akva2) [inline src/SIM/SIMmodal.C:80]: may as well check for non-null elem or some sca might get nervous.

**Review by akva2 — APPROVED** [review]
Two small suggestions.

File: src/SIM/SIMmodal.C (hunk 1 above)
Comment (kmokstad) [reply src/SIM/SIMmodal.C]: done

File: src/SIM/SIMmodal.C (hunk 2 above)
Comment (kmokstad) [reply src/SIM/SIMmodal.C]: ok, done that too. But there are any other places around where we don't do this check.

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=164 ifem-shellex=4 please

File: src/SIM/SIMmodal.C (hunk 2 above)
Comment (akva2) [reply src/SIM/SIMmodal.C]: yeah i'm aware. sometimes i've had to add checks to quell some warnings so just a defensive knee-jerk.

**Pull request merged**